### PR TITLE
irmin-pack: use pread/pwrite to avoid lseek calls

### PR DESCRIPTION
**Pull request by CraigFe**
Fixes https://github.com/mirage/irmin/issues/1009.

Depends on https://github.com/mirage/index/pull/176.

### CODE REVIEW

File: src/irmin-pack/IO.ml [hunk 1]
```diff
@@ -60,37 +60,38 @@ module Unix : S = struct
 
     let v fd = { fd; cursor = 0L }
```
Comment (icristescu) [inline src/irmin-pack/IO.ml:61]: I think we can remove the `cursor`, it was used to prevent unnecessary calls to `lseek`.

Comment (CraigFe) [reply src/irmin-pack/IO.ml:61]: Indeed. This `cursor` actually still exists in Index, even though there's no `lseek` being done there... Will remove from both.

**Comment by CraigFe** [comment]
> ( the benchmark is here it's an average of 5 runs, for nb of elements = 10000000 and element_size = 1000)

Those are really useful, thanks! Currently adapting them to this PR, and will report back with some results shortly.

**Comment by CraigFe** [comment]
Here are the benchmarking results:
image

| metric | value |
| - | - |
| nb_elements | 10,000,000 |
| element size (bytes) | 100 |
| repeats | 5 |

(Each run reads/writes `nb_elements` to the pack, in sequential or random order.)

In short, improvements across the board. Unsurprisingly, random access is more significantly improved.

**Comment by samoht** [comment]
It's nice to see such a speed-up!

**Pull request merged**